### PR TITLE
lib/deltas: Some misc declare-and-initialize porting

### PR DESCRIPTION
**Pull request by cgwalters**
Also some `glnx_fstatat_allow_noent()`.  Not specifically prep for anything.

### CODE REVIEW

**Comment by cgwalters** [comment]
This one should wait until after 2018.4 (the hotfix with https://github.com/ostreedev/ostree/pull/1508 )

**Comment by cgwalters** [comment]
Fixup :arrow_up: all green now.

File: src/libostree/ostree-repo-static-delta-core.c [hunk 1]
```diff
@@ -726,26 +708,16 @@ _ostree_repo_static_delta_query_exists (OstreeRepo                    *self,
...
-      if (errno == ENOENT)
-        {
-          *out_exists = FALSE;
-          return TRUE;
-        }
-      else
-        return glnx_throw_errno_prefix (error, "fstatat(%s)", superblock_path);
-    }
-  *out_exists = TRUE;
+  *out_exists = (errno == 0);
```
Comment (jlebon) [inline src/libostree/ostree-repo-static-delta-core.c]: Nice!

**Comment by jlebon** [comment]
@rh-atomic-bot r+ 1c53652

**Pull request closed without merging**